py-polars/src/series/buffers.rs

```diff
@@ -281,3 +277,135 @@ fn get_boolean_buffer_length_in_bytes(length: usize, offset: usize) -> usize {
         n_bytes + 1
     }
 }
+
+#[pymethods]
+impl PySeries {
+    /// Construct a PySeries from information about its underlying buffers.
+    #[staticmethod]
+    unsafe fn _from_buffers(
+        dtype: Wrap<DataType>,
+        data: Vec<PySeries>,
+        validity: Option<PySeries>,
+    ) -> PyResult<Self> {
+        let dtype = dtype.0;
+        let data = data.to_series();
+
+        match data.len() {
+            0 => {
+                return Err(PyTypeError::new_err(
+                    "`data` input to `from_buffers` must contain at least one buffer",
+                ));
+            },
+            1 if validity.is_none() => {
+                let data = data.into_iter().next().unwrap();
+                let s = data.strict_cast(&dtype).map_err(PyPolarsErr::from)?;
+                return Ok(s.into());
+            },
+            _ => (),
+        }
+
+        let validity = match validity {
+            Some(s) => {
+                let dtype = s.series.dtype();
+                if !dtype.is_bool() {
+                    return Err(PyTypeError::new_err(format!(
+                        "validity buffer must have data type Boolean, got {:?}",
+                        dtype
+                    )));
+                }
+                Some(series_to_bitmap(s.series).unwrap())
+            },
+            None => None,
+        };
+
+        let s = match dtype.to_physical() {
+            dt if dt.is_numeric() => {
+                let data = data.into_iter().next().unwrap();
+                with_match_physical_numeric_polars_type!(dt, |$T| {
+                    let data_buffer = series_to_buffer::<$T>(data);
+                    from_buffers_num_impl::<<$T as PolarsNumericType>::Native>(data_buffer, validity)?
+                })
+            },
+            DataType::Boolean => {
+                let data = data.into_iter().next().unwrap();
+                let data_buffer = series_to_bitmap(data)?;
```

Comment:
We typically call this `values`.

Reply:
Renamed here and in other relevant spots.